test/cachingoptimizer.jl

```diff
@@ -164,10 +164,26 @@ end
 
 @testset "CachingOptimizer constructor with optimizer" begin
     s = MOIU.MockOptimizer(ModelForMock{Float64}())
-    m = MOIU.CachingOptimizer(ModelForCachingOptimizer{Float64}(), s)
-    @test MOI.isempty(m)
-    @test MOIU.state(m) == MOIU.AttachedOptimizer
-    @test MOIU.mode(m) == MOIU.Automatic
+    model = ModelForCachingOptimizer{Float64}()
+    @testset "Empty model and optimizer" begin
+        m = MOIU.CachingOptimizer(model, s)
+        @test MOI.isempty(m)
+        @test MOIU.state(m) == MOIU.AttachedOptimizer
+        @test MOIU.mode(m) == MOIU.Automatic
+    end
+    @testset "Non-empty optimizer" begin
+        MOI.addvariable!(s)
```

Comment:
Tests shouldn't really share state like this, it defeats the readability point. It's better to regenerate `s` and `model`.

Reply:
Done :)